src/main/java/teammates/ui/webapi/GetNotificationAction.java

```diff
@@ -0,0 +1,53 @@
+package teammates.ui.webapi;
+
+import java.util.List;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.FieldValidator;
+import teammates.ui.output.NotificationData;
+import teammates.ui.output.NotificationsData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+
+/**
+ * Action: Get a list of notifications.
+ */
+public class GetNotificationAction extends Action {
+
+    @Override
+    AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    void checkSpecificAccessControl() throws UnauthorizedAccessException {
+        // Any user can get notifications as long as its parameters are valid
+    }
+
+    @Override
+    public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOperationException {
+        String targetUser = getRequestParamValue(Const.ParamsNames.NOTIFICATION_TARGET_USER);
+        //        boolean isFetchingAll = Boolean.parseBoolean(
+        //                getRequestParamValue(Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL));
+
+        String targetUserErrorMessage = FieldValidator.getInvalidityInfoForNotificationTargetUser(targetUser);
+        if (!targetUserErrorMessage.isEmpty()) {
+            throw new InvalidHttpRequestBodyException(targetUserErrorMessage);
+        }
+
+        List<NotificationAttributes> notificationAttributes =
+                logic.getNotificationsByTargetUser(NotificationTargetUser.valueOf(targetUser));
```

Comment:
Currently, the code only allows retrieving of notifications by target user type; we might need to add extra logic here to retrieve all notifications (either in this PR or future PRs), since it may needed for the admin notifications page.

Reply:
Sure, I think this is a good point, I will create another method for getting all notifications